### PR TITLE
fix: relations filtering on locales

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Fix relation findAvalaible 

Fix https://github.com/strapi/strapi/issues/21287
Fix https://github.com/strapi/strapi/issues/21286

### Why is it needed?

Describe the issue you are solving.

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: packages/core/content-manager/server/src/controllers/relations.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ const validateLocale = (sourceUid: UID.Schema, targetUid: UID.ContentType, local
 
   let validatedLocale = locale;
 
-  if (!targetModel || !isTargetLocalized) validatedLocale = undefined;
+  if (!targetModel && !isTargetLocalized) validatedLocale = undefined;
```
Comment (Feranchz) [inline packages/core/content-manager/server/src/controllers/relations.ts:96]: I think we can safely remove this line... I tried it locally and it was working.

We only use locale to filter if `isTargetLocalized` or `isSourceLocalized` is true. (depending on the case) So this looks like not necessary

Comment (alexandrebodin) [reply packages/core/content-manager/server/src/controllers/relations.ts:96]: agreed

**Comment by alexandrebodin** [comment]
Making an experimental here for QA @Feranchz cc @oiorain 
https://github.com/strapi/strapi/actions/runs/10898914294/job/30243144943

**Comment by oiorain** [comment]
@alexandrebodin regarding https://github.com/strapi/strapi/issues/21287
Im not sure this case is covered ny this fix but there is still this bug: 
- Create a CT in english
- create the french version: at first the relation is showing empty but after you save it appears with the same value as the english verison 
- same for a third japanese version

**Comment by oiorain** [comment]
confirmed this is not happening on the experimental! 
good job! 🚀

**Pull request merged**